sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuCast.scala

```diff
@@ -1093,8 +1060,19 @@ object GpuCast extends Arm {
               convertFixedLenTimestampOrNull(sanitizedInput, 4, "%Y"))))
 
       // handle special dates like "epoch", "now", etc.
-      val finalResult = specialDates.foldLeft(converted)((prev, specialDate) =>
-        specialTimestampOr(sanitizedInput, specialDate._1, specialDate._2, prev))
+      // `converted` will be closed in replaceSpecialDates. We wrap it with closeOnExcept in case
+      // of exception before replaceSpecialDates.
+      val finalResult = closeOnExcept(converted) { timeStampVector =>
+        val specialDates = Seq(DateUtils.EPOCH, DateUtils.NOW, DateUtils.TODAY,
```

Comment:
It seems expensive to replace the special dates with null with Spark 3.2 rather than just skip handling them at all. They will already be ignored by `is_timestamp`.

I will pull this PR locally today and experiment with it and come back with more detailed suggestions.